src/api/java/com/minecolonies/api/colony/IChunkmanagerCapability.java

```diff
@@ -0,0 +1,143 @@
+{
+    /**
+     * Get the chunkStorage at a certain location.
+     * @param chunkX the x chunk location.
+     * @param chunkZ the z chunk location.
+     * @return the storage or null.
+     */
+    @Nullable
+    ChunkLoadStorage getChunkStorage(int chunkX, int chunkZ);
+
+    /**
+     * Add a new chunkStorage.
+     * @param chunkX chunkX the x chunk location.
+     * @param chunkZ chunkX the z chunk location.
+     * @param storage the new to add or update.
+     * @return true if override else false.
+     */
+    boolean addChunkStorage(int chunkX, int chunkZ, ChunkLoadStorage storage);
+
+    /**
+     * Get all chunk storages for serialization.
+     * @return the storages.
+     */
+    Map<ChunkPos, ChunkLoadStorage> getAllChunkStorages();
+
+    /**
+     * The implementation of the colonyTagCapability.
+     */
+    class Impl implements IChunkmanagerCapability
+    {
+        /**
+         * Map of chunkPos to chunkLoadStorage.
+         */
+        private final Map<ChunkPos, ChunkLoadStorage> chunkStorages = new HashMap<>();
+
+        @Nullable
+        @Override
+        public ChunkLoadStorage getChunkStorage(final int chunkX, final int chunkZ)
+        {
+            return chunkStorages.get(new ChunkPos(chunkX, chunkZ));
+        }
+
+        @Override
+        public boolean addChunkStorage(final int chunkX, final int chunkZ, final ChunkLoadStorage storage)
+        {
+            final ChunkLoadStorage existingStorage = chunkStorages.get(new ChunkPos(chunkX, chunkZ));
+            if (existingStorage == null)
+            {
+                chunkStorages.put(new ChunkPos(chunkX, chunkZ), storage);
+                return false;
+            }
+            else
+            {
+                existingStorage.merge(storage);
+                return true;
+            }
+        }
+
+
+        @Override
+        public Map<ChunkPos, ChunkLoadStorage> getAllChunkStorages()
+        {
+            return chunkStorages;
+        }
+    }
+
+    /**
+     * The storage class of the capability.
+     */
+    class Storage implements Capability.IStorage<IChunkmanagerCapability>
+    {
+        @Override
+        public NBTBase writeNBT(@NotNull final Capability<IChunkmanagerCapability> capability, @NotNull final IChunkmanagerCapability instance, @Nullable final EnumFacing side)
+        {
+            final NBTTagCompound compound = new NBTTagCompound();
+            compound.setTag(TAG_ALL_CHUNK_STORAGES, instance.getAllChunkStorages().entrySet().stream().map(entry -> write(entry.getKey(), entry.getValue())).collect(NBTUtils.toNBTTagList()));
+            return compound;
+        }
+
+        @Override
+        public void readNBT(@NotNull final Capability<IChunkmanagerCapability> capability, @NotNull final IChunkmanagerCapability instance,
+          @Nullable final EnumFacing side, @NotNull final NBTBase nbt)
+        {
+            if(nbt instanceof NBTTagCompound && ((NBTTagCompound) nbt).hasKey(TAG_ID))
```

Comment:
### Errors:
- Condition nbt instanceof NBTTagCompound is always false
- Condition nbt instanceof NBTTagCompound && ((NBTTagCompound) nbt).hasKey(TAG_ID) is always false